### PR TITLE
Admin edit prizes

### LINKED ISSUE #140: Admins can edit prizes

### PR DESCRIPTION
**Pull request by jdg006**
## Description

Admins can now edit prizes from the admin dashboard by clicking the edit button next to the title of the prize. 

Fixes #140

## Developer Checklist
*You can mark items off by replacing `[ ]` with `[x]` or in the rendered
markdown, you can simply check the checkbox.*
- [x] I have read and followed the [contribution guidelines](
      https://github.com/crawfoal/greensteps/blob/master/CONTRIBUTING.md).
- [x] All specs and linters pass (including Code Climate)
- [x] I have followed the style of nearby code
- [x] The build on Semaphore is successful
- [x] I have added tests for my changes.
- [x] I have gone through my changes by hand, either in the browser or terminal,
      which ever makes sense
- [x] I have [internationalized](http://guides.rubyonrails.org/i18n.html) my
      changes

### CODE REVIEW

**Comment by jdg006** [comment]
The spec that is not passing on semaphoreci passes locally. Let me know what I need to do here to get it passing and I'll get right to it.

**Comment by jdg006** [comment]
I had to change some of the paths I was using because I noticed that the paths had been changed on the master branch's  config file. I think they are all good now though. Let me know what you think and if you want me to change anything.

File: app/controllers/admins/prizes_controller.rb [hunk 1]
```diff
@@ -12,6 +12,21 @@ def create
       end
     end
 
+    def edit
+      @prize = Prize.find(params[:id])
```
Comment (crawfoal) [inline app/controllers/admins/prizes_controller.rb]: We're using the decent_exposure gem in this project. In their words, decent_exposure is a "helper for creating declarative interfaces in controllers". They have a brief explanation of how to use the gem in [the API section of their README](https://github.com/hashrocket/decent_exposure#api) (for now, it would suffice to stop reading when you get to "Under the Hood").

We should use the `prize` method to access the appropriate prize instead of looking up in each controller action. Because `prize` is defined as a helper method by decent_exposure, it also available in the views.

File: config/database.yml [hunk 2]
```diff
@@ -58,7 +58,7 @@ development:
 test:
   <<: *default
   database: greensteps_test
-
+
```
Comment (crawfoal) [inline config/database.yml]: Nitpick - delete the whitespace

File: db/schema.rb [hunk 3]
```diff
@@ -48,10 +48,11 @@
     t.string "name"
     t.string "resource_type"
     t.integer "resource_id"
-    t.datetime "created_at"
-    t.datetime "updated_at"
+    t.datetime "created_at", null: false
```
Comment (crawfoal) [inline db/schema.rb]: I don't think the changes to this file will hurt anything, but in a Rails app you shouldn't see any changes to the schema file without corresponding migrations. Do you happen to know how these changes got in here?

File: spec/controllers/admins/prizes_controller_spec.rb [hunk 4]
```diff
@@ -13,6 +15,56 @@
     end
   end
 
+  describe 'GET edit' do
```
Comment (crawfoal) [inline spec/controllers/admins/prizes_controller_spec.rb]: It is important to have full stack tests for each major feature of the app. We should have feature specs that cover the functionality tested here.

In order to avoid having multiple tests that test the same thing, I like to test the "happy path" in a feature spec and then the "sad path" (errors/failures or other edge cases) in a controller test (this is a common approach in Rails testing).

**Comment by crawfoal** [comment]
Also, don't worry about flaky test that are unrelated to what you're working on. I fixed those in the PR I'm working on, so once I get that finished and merged in, you could pull those changes into your branch if the flaky tests are causing issues for you.

File: app/views/admins/prizes/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+<h2><%= t('.header') %></h2>
+<%= render 'form' %>
```
Comment (crawfoal) [inline app/views/admins/prizes/edit.html.erb]: Nitpick - add a newline

**Comment by crawfoal** [comment]
Also, can you add some basic styling? A couple things that stood out to me were
  (1) it would be good to apply the prize form styles to edit form too
  (2) maybe play around with the position of the edit button - would it look better if it was right aligned and had a little left padding?

File: db/schema.rb (hunk 3 above)
Comment (jdg006) [reply db/schema.rb]: These changes got in here after I ran rubocop.  Is that normal or was it because of something else I did?

**Comment by jdg006** [comment]
I think I have corrected everything. Let me know if I need to add anything or if I need to make further corrections.

File: app/views/admins/prizes/edit.html.erb (hunk 5 above)
Comment (crawfoal) [reply app/views/admins/prizes/edit.html.erb]: ✔️

File: db/schema.rb (hunk 3 above)
Comment (jdg006) [reply db/schema.rb]: I reverted the schema back to what it previously was and ran rubocop and the tests again and it was not rubocop that made the changes, it was the specs. As the tests were running I watched the schema.rb file go from 113 lines to 116.  Let me know what you want me to do with this. I can revert the schema.rb file back to what it was and push it to my feature branch to get rid of these changes.

Comment (crawfoal) [reply db/schema.rb]: > I can revert the schema.rb file back to what it was and push it to my feature branch to get rid of these changes.

That sounds good, thanks.

I'd like to figure out how we can fix this issue for you though. If you work on other things in this project, I'd hate for you to have to deal with this issue every time you run the specs. After you revert the file back, can you try running `bin/rails db:reset` and then running the specs again?

Comment (jdg006) [reply db/schema.rb]: I just checked my psql version and it's 9.3. I see the project calls for 9.6. From what I could glean from stack overflow this is likely the problem. Unfortunately I am struggling to upgrade psql to version 9.6 because I am using a cloud 9 environment and do not have access to certain files required to upgrade psql. This might just have to be an issue that I deal with any time I make pull-requests.

Comment (crawfoal) [reply db/schema.rb]: Bummer 🙁 Are you using Cloud9 because you have a Windows operating system?

**Comment by crawfoal** [comment]
@jdg006 Thank you for your work on this!
